src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsService.kt

```diff
@@ -36,4 +42,18 @@ class NonAssociationsService(
   private fun persistNonAssociation(nonAssociation: NonAssociationJPA): NonAssociationJPA {
     return nonAssociationsRepository.save(nonAssociation)
   }
+
+  fun mergePrisonerNumbers(oldPrisonerNumber: String, newPrisonerNumber: String): List<NonAssociationJPA> {
+    log.info("Replacing prisoner number $oldPrisonerNumber to $newPrisonerNumber")
+
+    val nonAssociationList1 = nonAssociationsRepository.findAllByFirstPrisonerNumber(oldPrisonerNumber).map {
+      it.copy(firstPrisonerNumber = newPrisonerNumber)
```

Comment:
if `firstPrisonerNumber`/`secondPrisonerNumber` are changed to be `var` (as they should as they need to be updated now) you shouldn't need to `copy()`, you could just update the instance with `it.firstPrisonerNumber = newPrisonerNumber`

(so you don't need to make the entity a data class which could lead to unexpected issues with JPA)